homeassistant/components/sensor/forecast.py

```diff
@@ -129,75 +122,72 @@ def unit_of_measurement(self):
     @property
     def unit_system(self):
         """Return the unit system of this entity."""
-        return self._unit_system
+        return self.forecast_data.unit_system
+
+    def update_unit_of_measurement(self):
+        """Update units based on unit system."""
+        unit_index = {
+            'si': 1,
+            'us': 2,
+            'ca': 3,
+            'uk': 4,
+            'uk2': 5
+        }.get(self.unit_system, 1)
+        self._unit_of_measurement = SENSOR_TYPES[self.type][unit_index]
 
-    # pylint: disable=too-many-branches,too-many-statements
     def update(self):
         """Get the latest data from Forecast.io and updates the states."""
-        import forecastio
+        # Call the API for new forecast data. Each sensor will re-trigger this
+        # same exact call, but thats fine. We cache results for a short period
+        # of time to prevent hitting API limits. Note that forecast.io will
+        # charge users for too many calls in 1 day, so take care when updating.
+        self.forecast_data.update()
```

Comment:
Looking at the SDK we call down into, calling update makes an API call to forecast.io which gets us everything we need. We then call update on each individual component. With those calls, the object we get back from SDK has a method to see if the data (minutely, hourly, etc.) already exists in the json object it got back from the API. If it doesn't, then it'll make a new call to forecast.io. Looking at it though, that new call is hitting the same endpoint for the same data, so for our simple implementation, there is no reason it wouldn't already have all the data it needs.

Those methods also have rate limits on them from our side of things. (2 minutes) Since the update all call occurs first and we cache the result for 2 minutes, it's probably unnecessary to throttle all of those update calls, but a nice safeguard in my opinion.